tools/RepositoryManagement/DevHome.RepositoryManagement/Services/RepositoryManagementDataAccessService.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using DevHome.Common.Extensions;
+using DevHome.Database;
+using DevHome.Database.DatabaseModels.RepositoryManagement;
+using DevHome.RepositoryManagement.ViewModels;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Query;
+using Microsoft.Extensions.Hosting;
+using Serilog;
+
+namespace DevHome.RepositoryManagement.Services;
+
+public class RepositoryManagementDataAccessService
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(RepositoryManagementDataAccessService));
+
+    private readonly IHost _host;
+
+    public RepositoryManagementDataAccessService(IHost host)
+    {
+        // Store the host to make dbContext and RepositoryManagementItemViewModel
+        // dbContext can not be passed in because RepositoryManagementDataAccessService is singleton
+        // and dbContext needs to be scoped.
+        // RepositoryManagementItemViewModel can not be passed in because multiple are made.
+        // The best solution is to make factories for DevHomeContext and RepositoryManagementItemViewModel.
+        // Might be in a future change.
+        _host = host;
+    }
+
+    public void AddRepository(string repositoryName, string cloneLocation)
+    {
+        Repository newRepo = new();
+        newRepo.RepositoryName = repositoryName;
+        newRepo.RepositoryClonePath = cloneLocation;
+
+        RepositoryMetadata newMetadata = new();
+        newMetadata.Repository = newRepo;
+        newMetadata.RepositoryId = newRepo.RepositoryId;
+        newMetadata.IsHiddenFromPage = false;
+
+        newRepo.RepositoryMetadata = newMetadata;
+
+        var dbContext = _host.GetService<DevHomeDatabaseContext>();
+        dbContext.Add(newRepo);
+        dbContext.Add(newMetadata);
+        dbContext.SaveChanges();
```

Comment:
i'm confused how these are tied to the implicit transaction?

what if another thread is in the middle of adding a new repo - does this SaveChanges commit that (half-done) add?

Reply:
DbContext is a transient.  That means every call to .GetService<DevHomeDatabaseContext>() gets a new context and this is how multi-threaded is handled.  

Now, the scenario two threads with the same context isn't handled in this PR.  I will add more try/catches in the future.

> does this SaveChanges commit that (half-done) add
.SaveChanges() looks at any entities that were modified.  Entities have an internal state that tells EF what has changed.

the .Add on line 50 changes the internal state of newMetadata to "Added".  .SaveChanges pick up this and will add the entity to the database. :)